### PR TITLE
Fix fast jit coverity issues

### PR DESCRIPTION
**Pull request by wenyongh**

### CODE REVIEW

File: core/iwasm/fast-jit/cg/x86-64/jit_codegen_x86_64.cpp [hunk 1]
```diff
@@ -186,41 +186,45 @@ jit_codegen_interp_jitted_glue(void *exec_env, JitInterpSwitchInfo *info,
         }                                                       \
     } while (0)
 
-#define ASSERT_I32_REG_NO(no) \
-    bh_assert(no >= 0 && (uint32)no < sizeof(regs_i32) / sizeof(regs_i32[0]))
+#define CHECK_I32_REG_NO(no)                                                   \
+    do {                                                                       \
+        if (!(no >= 0 && (uint32)no < sizeof(regs_i32) / sizeof(regs_i32[0]))) \
```
Comment (lum1n0us) [inline core/iwasm/fast-jit/cg/x86-64/jit_codegen_x86_64.cpp:191]:
```suggestion
        if ((uint32)no >= sizeof(regs_i32) / sizeof(regs_i32[0]))) \
```
just realized, may be we can simplify the condition.

File: core/iwasm/fast-jit/cg/x86-64/jit_codegen_x86_64.cpp [hunk 2]
```diff
@@ -5704,54 +5754,60 @@ lower_callnative(JitCompContext *cc, x86::Assembler &a, JitInsn *insn)
...
+                    int32 arg_reg_no = jit_reg_no(arg_reg);
+                    CHECK_I64_REG_NO(arg_reg_no);
+                    mov_r_to_r_i64(a, reg_no, arg_reg_no);
                 }
                 break;
             }
             case JIT_REG_KIND_F32:
             {
-                ASSERT_F32_REG_NO(floatpoint_reg_index);
+                CHECK_F32_REG_NO((int32)floatpoint_reg_index);
```
Comment (lum1n0us) [inline core/iwasm/fast-jit/cg/x86-64/jit_codegen_x86_64.cpp:5787]: maybe `int8 floatpoint_reg_index = 0;` -> `int floatpoint_reg_no = 0`

**Review by lum1n0us — COMMENTED** [review]
Btw, `if` maybe a little heavy when we are sure `register allocation` always give valid register numbers?

File: core/iwasm/fast-jit/cg/x86-64/jit_codegen_x86_64.cpp (hunk 1 above)
Comment (wenyongh) [reply core/iwasm/fast-jit/cg/x86-64/jit_codegen_x86_64.cpp:191]: OK, sounds better.

**Comment by wenyongh** [comment]
> Btw, `if` maybe a little heavy when we are sure `register allocation` always give valid register numbers?

It is just for debug version, in release version, CHECK_REG_NO is defined as (void)0.

**Pull request closed without merging**